clang/docs/UsersManual.rst

```diff
@@ -1315,6 +1319,34 @@ with its corresponding `Wno-` option.
 Note that when combined with :option:`-w` (which disables all warnings),
 disabling all warnings wins.
 
+.. _warning_suppression_mappings:
+
+Controlling Diagnostics via Suppression Mappings
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Warning suppression mappings enable users to suppress Clang's diagnostics in a
+per-file granular manner. Enabling enforcement of diagnostics in specific parts
+of the project, even if there are violations in some headers.
```

Comment:
Remove comma: project even if there
Reword: some headers -> some headers which cannot be modified.

Reply:
> Reword: some headers -> some headers which cannot be modified.

But this covers headers even inside the project (i.e. the incremental cleanup of a new warning). So I feel like `which cannot be modified.` adds a non-existing limitation.

Reply:
That's true, but I think we want to guide users towards fixing diagnostics rather than suppressing them. Users will use any suppression mechanisms we provide regardless of how we document it, so I don't think it's harmful to imply this is more about cases where you can't modify the code. But I also don't feel strongly. WDYT?